tm/src/main/java/io/seata/tm/DefaultTransactionManager.java

```diff
@@ -51,7 +52,7 @@ public String begin(String applicationId, String transactionServiceGroup, String
         request.setTransactionName(name);
         request.setTimeout(timeout);
         GlobalBeginResponse response = (GlobalBeginResponse) syncCall(request);
-        if (response.getResultCode() == ResultCode.Failed) {
+        if (Objects.isNull(response) || response.getResultCode() == ResultCode.Failed) {
```

Comment:
need to submit UT ?

Reply:
Under what circumstances will NPE occur here?

Reply:
when seats-sample project runs,, but not start seats-server.sh script, occurs! 
<img width="1230" alt="image" src="https://github.com/seata/seata/assets/48004670/fc5dfa05-abc0-4047-ac37-98a7c76789fc">

Reply:
而且堆栈是54行，你修改的是55行，你应该提交一个issue，而不是一个pr，一个记录完整异常描述和seata版本的issue

Moreover, the stack trace is at line 54 and you made a modification at line 55. You should submit an issue instead of a pull request, which includes the complete description of the exception and the version of Seata.

Reply:
ok